### ISSUE TITLE
Support more possible delimiter and quote characters

### ISSUE THREAD

**Issue by owen-synergy** [yehoshuadimarsky/bcpandas#12]
Hi Josh, I'm not sure if this is possible, but can BCP and Pandas to_csv support more delimiter and quote characters than just the ones you've specified so far? I'm possibly trying to push the limits of bcpandas too far by writing DataFrames with big json blob elements (I know you don't recommend using it with messy text data) but I didn't know if there was still potential to improve it in that regard.

Happy to try creating a PR if you think there's still potential improvements here.

**Comment by yehoshuadimarsky** [comment]
Fair point. Technically, I think every single character is fair game to be used as a delimiter or quote character, as long as it doesn't appear in the actual data. We can (and do) check for this `constants.py`. So we could change it to allow for all characters in ASCII 32-127. I was just unsure how this would actually work in real life as it is a bit unorthodox, and also I've never tried it. But we can try it.

**Comment by owen-synergy** [comment]
I'll have to double check the BCP docs again... I think I remember it having to be a visible character or a few others (maybe that is the 32 - 127 range...). Using other characters for quotes does seem a bit funny to me too... I didn't see anything in the Pandas documentation as what characters it would allow for that.

I'll see if I can find some time to experiment. I was going to try and extract the set of all characters that appear in the DataFrame to check if there's two left over that can be used.

**Comment by yehoshuadimarsky** [comment]
Taking a stab now at working through it... Here's the link for Microsoft docs about terminators 
https://docs.microsoft.com/en-us/sql/relational-databases/import-export/specify-field-and-row-terminators-sql-server

**Comment by owen-synergy** [comment]
Nice, I made this code snippet to grab all characters from all strings in a DataFrame, not sure if there's a more efficient way...

```python
all_chars = set()
for col in df.columns:
    try:
        all_chars.update(set(df[col].str.cat(sep="")))
    except AttributeError:
        pass
```

**Comment by yehoshuadimarsky** [comment]
There is a better way - it's easier to just go through all the possible valid ASCII values, and check each one that it's not in the DataFrame using the same code we currently use 

https://github.com/yehoshuadimarsky/bcpandas/blob/d6b16141e40bbabc28ea12ed21ed05eaaf999c51/bcpandas/constants.py#L64-L68

**Comment by puruzio** [comment]
Any chance to add | (pipeline char) as the delimiter? My data team seems to like including the common delimiter characters in the data, and I keep getting this error.

```
bcpandas.constants.BCPandasValueError: Data contains all of the possible quote characters ('"', "'", '`', '~'),
cannot use BCP to import it. Replace one of the possible quote characters in
your data, or use another method besides bcpandas.

``````

**Comment by yehoshuadimarsky** [comment]
It already is one of the delimiter options https://github.com/yehoshuadimarsky/bcpandas/blob/db37a2c1ffba7f19a01019b833ba526622ae577c/bcpandas/constants.py#L38
